app/concepts/reservations/use_cases/create.rb

```diff
@@ -0,0 +1,71 @@
+module Reservations
+  module UseCases
+    class Create
+      ReservationInvalidError = Class.new(StandardError)
+
+      attr_reader :repository, :params, :ticket_desk_id
+
+      def initialize(params:, ticket_desk_id:, repository: Reservations::Repository.new)
+        @repository = repository
+        @params = params
+        @ticket_desk_id = ticket_desk_id
+      end
+
+      def call
+        Reservation.transaction do
+          repository.create(reservation_params).tap do |reservation|
+            raise ReservationInvalidError, "Couldn't create reservation" if reservation.invalid?
+
+            Tickets::UseCases::CreateForReservation.new(
+              reservation: reservation,
+              screening: screening,
+              tickets_params: tickets_params
+            ).call
+
+            CancelReservationsJob.set(wait: 30.seconds).perform_later(reservation.id) if ticket_desk.online
```

Comment:
> I would like to perform the async job at certain time described as expires_at and check if the reservation changed its status

Consider that maybe this code will be useful here:
```ruby
CancelReservationsJob.perform_at(reservation.expires_at, reservation.id)
```

Check this article
https://github.com/mperham/sidekiq/wiki/Scheduled-Jobs